compiler/ir/backend.wasm/src/org/jetbrains/kotlin/backend/wasm/lower/BuiltInsLowering.kt

```diff
@@ -6,40 +6,139 @@
+                    val newSymbol = symbols.equalityFunctions[lhsType]
+                    if (newSymbol != null) {
+                        return irCall(call, newSymbol)
+                    }
+                }
+                if (lhs.isNullConst()) {
+                    return builder.irCall(symbols.refIsNull).apply { putValueArgument(0, rhs) }
+                }
+                if (rhs.isNullConst()) {
+                    return builder.irCall(symbols.refIsNull).apply { putValueArgument(0, lhs) }
+                }
+                if (!lhsType.isNullable()) {
+                    return irCall(call, lhsType.findEqualsMethod().symbol, argumentsAsReceivers = true)
+                }
+                return irCall(call, symbols.nullableEquals)
+            }
+
+            irBuiltins.eqeqeqSymbol -> {
                 val type = call.getValueArgument(0)!!.type
-                val newSymbol = symbols.equalityFunctions[type]
-                    ?: error("Unsupported equality operator with type: ${type.render()}")
+                val newSymbol = symbols.equalityFunctions[type] ?: symbols.floatEqualityFunctions[type] ?: symbols.refEq
                 return irCall(call, newSymbol)
             }
-            in symbols.irBuiltInsToWasmIntrinsics.keys -> {
-                val newSymbol = symbols.irBuiltInsToWasmIntrinsics[symbol]!!
+
+            irBuiltins.checkNotNullSymbol -> {
+                return irCall(call, symbols.ensureNotNull).also {
+                    it.putTypeArgument(0, call.type)
+                }
+            }
+            in symbols.comparisonBuiltInsToWasmIntrinsics.keys -> {
+                val newSymbol = symbols.comparisonBuiltInsToWasmIntrinsics[symbol]!!
                 return irCall(call, newSymbol)
             }
+
+            // TODO: Implement
+            irBuiltins.noWhenBranchMatchedExceptionSymbol ->
+                return builder.irCall(symbols.wasmUnreachable, irBuiltins.nothingType)
+
+            // TODO: Implement
+            irBuiltins.illegalArgumentExceptionSymbol ->
+                return builder.irCall(symbols.wasmUnreachable, irBuiltins.nothingType)
+
+            irBuiltins.dataClassArrayMemberHashCodeSymbol -> {
+                // TODO: Implement
+                return builder.irComposite {
+                    +call.getValueArgument(0)!!
+                    +irInt(7777)
+                }
+            }
+            irBuiltins.dataClassArrayMemberToStringSymbol -> {
+                // TODO: Implement
+                return builder.irCall(symbols.anyNtoString).apply {
+                    putValueArgument(0, call.getValueArgument(0))
+                }
+            }
+        }
+
+        val nativeInvoke = isNativeInvoke(call)
+        if (nativeInvoke != null) {
+            return irCall(call, symbols.functionNInvokeMethods[nativeInvoke])
         }
+
         return call
     }
 
+    private fun isNativeInvoke(call: IrCall): Int? {
```

Comment:
Let's rename it to something like`getKotlinFunctionInvokeArity`

Reply:
Renamed